apps/dsaparam.c

```diff
@@ -66,9 +66,8 @@ const OPTIONS dsaparam_options[] = {
 int dsaparam_main(int argc, char **argv)
 {
     ENGINE *e = NULL;
-    DSA *dsa = NULL;
     BIO *in = NULL, *out = NULL;
-    EVP_PKEY *pkey = NULL;
+    EVP_PKEY *params, *pkey = NULL;
```

Comment:
gcc in a Travis run correctly pointed out that `params` needs to be initialized (to `NULL`).
Then the crash disappears, and everything seems to work fine :)